prov/tcp/src/tcpx.h

```diff
@@ -117,8 +143,14 @@ struct poll_fd_mgr {
 	int			max_nfds;
 };
 
+struct tcpx_conn_req {
+	struct ofi_ctrl_hdr hdr;
+	uint16_t cm_data_sz;
```

Comment:
I think you can just use the seg_size field from the ctrl_hdr.  The documentation in the header files isn't a requirement, just a guideline for how specific fields can be used.  The actual use depends on the provider and what underlying communication protocol is in use.